config/passport.js

```diff
@@ -0,0 +1,75 @@
+const passport = require('passport');
+const JwtStrategy = require('passport-jwt').Strategy;
+const ExtractJwt = require('passport-jwt').ExtractJwt;
+const mongoose = require('mongoose');
+var user = require('./../models/user');
+mongoose.model('user');
+const keys = require('./keys');
+
+const GoogleStrategy = require('passport-google-oauth20').Strategy;
+
+passport.serializeUser((user, done) => {
+  done(null, user.id);
+});
+
+passport.deserializeUser((id, done) => {
+  User.findById(id).then((user) => {
+    done(null, user);
+  });
+});
+
+
+
+const opts = {};
+opts.jwtFromRequest = ExtractJwt.fromAuthHeaderAsBearerToken();
+opts.secretOrKey = keys.secret;
+
+module.exports = passport => {
+  passport.use(
+    new GoogleStrategy({
+      // options for google strategy
+      clientID: keys.google.clientID,
+      clientSecret: keys.google.clientSecret,
+      callbackURL: '/api/users/google/redirect'
```

Comment:
It might be easier to create a configuration JSON file that gets imported and imported just to keep common configuration in a single place.